src/hooks/useEditorView.ts

```diff
@@ -1,56 +1,33 @@
-import type { EditorState, Transaction } from "prosemirror-state";
+import { Schema } from "prosemirror-model";
+import { EditorState } from "prosemirror-state";
+import type { Plugin, Transaction } from "prosemirror-state";
 import { EditorView } from "prosemirror-view";
-import type { DirectEditorProps } from "prosemirror-view";
-import { useLayoutEffect, useState } from "react";
+import type { EditorProps } from "prosemirror-view";
+import { useLayoutEffect, useMemo, useState } from "react";
 import { flushSync } from "react-dom";
 
-import { useForceUpdate } from "./useForceUpdate.js";
+import type { EditorContextValue } from "../contexts/EditorContext.js";
 
-function withFlushedUpdates<This, T extends unknown[]>(
-  fn: (this: This, ...args: T) => void
-): (...args: T) => void {
-  return function (this: This, ...args: T) {
-    flushSync(() => {
-      fn.call(this, ...args);
-    });
-  };
-}
+import { useComponentEventListeners } from "./useComponentEventListeners.js";
 
-function defaultDispatchTransaction(this: EditorView, tr: Transaction) {
-  this.updateState(this.state.apply(tr));
-}
+const EMPTY_SCHEMA = new Schema({
+  nodes: {
+    doc: { content: "text*" },
+    text: { inline: true },
+  },
+});
 
-type EditorStateProps =
-  | {
-      state: EditorState;
-    }
-  | {
-      defaultState: EditorState;
-    };
```

Comment:
I think it's also easier to follow the logic of warning when `defaultValue` and `value` are both specified than it is to follow conditional type narrowing.

Reply:
I agree, but I'm sad about the loss of DX. Would we be open to keeping this type for the public props, and casting it internally to make it simpler for us?

Reply:
Do you think it's really bad? Folks will still get autocomplete telling them that these both exist, and I think they're not likely to try to specify both.

Reply:
This matches what React itself does with `<input />`, if that's any reassurance.

Reply:
I don't think it's _really_ bad, for sure. I just think it's very nice when you can specify precisely the type you mean, especially for a third-party lib, to remove any potential for confusion, and since we already had it, I'm sad to see it go

Reply:
I think it'll be messy to try to have it both ways. I value the readability here and doubt there's much of any DX hit to users.

Reply:
I'll try once more, or feel free to propose something.

Reply:
I gave it one more effort, but I think it's awkward. Should passing `state={undefined} defaultState={something}` be allowed by typing? I kinda think yes.

Also, if we type them as a union then checking for user error for the non-TypeScript users becomes awkward. TypeScript tries to tell us that we can't, that that can't happen, but of course it can.

Reply:
When I say "should passing ... be allowed" what I mean is that while TypeScript will allow you to, say, spread an object with extra props into a component, that might cause `"foo" in props` and `props.foo === undefined` to be different. I think the latter is the appropriate check that we should be making in the implementation.

Reply:
You, too. Great questions.